src/v/security/scram_credential.h

```diff
@@ -47,6 +62,7 @@ class scram_credential
     bytes _server_key;
     bytes _stored_key;
     int _iterations{0};
+    std::optional<acl_principal> _principal;
```

Comment:
I need to fix the versioned serialisation for this.

Reply:
Is this sufficient?
```
-  : public serde::envelope<scram_credential, serde::version<0>> {
+  : public serde::
+      envelope<scram_credential, serde::version<1>, serde::compat_version<0>> {
```

Reply:
yeh looks good

Reply:
I've avoided bumping the version by not serialising the principal.